src/CodeGeneration.Roslyn/RichGenerationResult.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Andrew Arnott. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace CodeGeneration.Roslyn
+{
+    using System.Diagnostics;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+    /// <summary>
+    /// Contains <see cref="CompilationUnitSyntax"/> additions generated by the <see cref="IRichCodeGenerator"/>.
+    /// </summary>
+    public struct RichGenerationResult
+    {
+        /// <summary>
+        /// Creates <see cref="RichGenerationResult"/> with provided arguments as property values.
+        /// </summary>
+        /// <param name="members">Assigned to <see cref="Members"/>.</param>
+        /// <param name="usings">Assigned to <see cref="Usings"/>.</param>
+        /// <param name="attributeLists">Assigned to <see cref="AttributeLists"/>.</param>
+        /// <param name="externs">Assigned to <see cref="Externs"/>.</param>
+        [DebuggerStepThrough]
+        public RichGenerationResult(
+            SyntaxList<MemberDeclarationSyntax> members,
+            SyntaxList<UsingDirectiveSyntax> usings = default,
+            SyntaxList<AttributeListSyntax> attributeLists = default,
+            SyntaxList<ExternAliasDirectiveSyntax> externs = default)
+        {
```

Comment:
Instead of these ctors, which will be troublesome to version as we add more members, why not give the properties public setters and let them set whatever ones they want?